homeassistant/components/go2rtc/server.py

```diff
@@ -66,9 +64,30 @@ async def start(self) -> None:
         )
 
         self._hass.async_create_background_task(
-            _log_output(self._process), "Go2rtc log output"
+            self._log_output(self._process), "Go2rtc log output"
         )
 
+        try:
+            async with asyncio.timeout(_SETUP_TIMEOUT):
+                await self._startup_complete.wait()
+        except TimeoutError as err:
+            msg = "Go2rtc server didn't start correctly"
+            _LOGGER.exception(msg)
+            await self.stop()
+            raise HomeAssistantError("Go2rtc server didn't start correctly") from err
+
+    async def _log_output(self, process: asyncio.subprocess.Process) -> None:
+        """Log the output of the process."""
+        assert process.stdout is not None
+
+        async for line in process.stdout:
+            msg = line[:-1].decode().strip()
+            _LOGGER.debug(msg)
+            if not self._startup_complete.is_set() and msg.endswith(
+                _SUCCESSFUL_BOOT_MESSAGE
+            ):
+                self._startup_complete.set()
```

Comment:
Parsing the log seems fragile, and we need to trust the log we wait for only happens on success and only after the endpoint is up.
As discussed, I suggest to just try to repeatedly reach the endpoint instead of parsing log output.

Reply:
Let's change this in a follow-up

Reply:
We leave it for now. We can always replace it with a better solution in the future